lib/containers/common.pm

```diff
@@ -262,4 +262,39 @@ sub test_rpm_db_backend {
     }
 }
 
+sub check_containers_connectivity {
+    my $runtime = shift;
+    record_info "connectivity", "Checking that containers can connect to the host, to each other and outside of the host";
+    my $container_name = 'sut_container';
+
+    # Run container in the background
+    assert_script_run "$runtime pull " . registry_url('alpine');
+    assert_script_run "$runtime run -id --rm --name $container_name -p 1234:1234 " . registry_url('alpine') . " sleep 30d";
+    my $container_ip = container_ip $container_name, $runtime;
+
+    # Connectivity to host check
+    my $container_route = container_route($container_name, $runtime);
+    assert_script_run "ping -c3 " . $container_route;
+    assert_script_run "$runtime run --rm " . registry_url('alpine') . " ping -c3 " . $container_route;
+
+    # Cross-container connectivity check
+    assert_script_run "ping -c3 " . $container_ip;
+    assert_script_run "$runtime run --rm " . registry_url('alpine') . " ping -c3 " . $container_ip;
+
+    # Outisde IP connectivity check
+    zypper_call('in bind-utils') if (!is_transactional() && script_run("which dig") != 0);
+    if (script_run("which dig") == 0) {
+        my $remote_ip = script_output 'dig +short google.com';
+        assert_script_run "wget $remote_ip";
+        assert_script_run "$runtime run --rm " . registry_url('alpine') . " wget $remote_ip";
+    }
+
+    # Outside IP+DNS connectivity check
+    assert_script_run "wget google.com";
+    assert_script_run "$runtime run --rm " . registry_url('alpine') . " wget google.com";
```

Comment:
Some observations:
- looks like a if..else thing here
- Why do you need the first `assert_script_run`? the host is tested already in this point anyway

Reply:
It's not `if..else` - I need to check that the connectivity (including DNS) works. To be sure that the problem is container-specific and not host-related, I decided to run the command firstly from the host and then from the container.

Reply:
IMO this doesnt make sense. But i wont object if others are good with it. thanks @pdostal

Reply:
I agree with both of you. I would actually compare somehow both outputs, to make sure that it doesn't return some 50X error or anything weird.

I would approach this differently, instead of `wget`, I would use `nslookup`, since we want to check DNS resolution only.
What about something like this?

```
my $server1 = script_output 'nslookup google.com | grep Server';
my $server2 = script_output 'podman run --rm alpine nslookup google.com | grep Server';
die 'some error message' if ($server1 ne $server2);
```

Reply:
Well but `nslookup` as well as `dig` are in `bind-utils` package sometimes not installed by default neither in (open)SUSE nor in Alpine container. That's why I use `wget` - it's everywhere.

Reply:
`nslookup` is in Alpine. The commands I posted were tested locally.

Reply:
ah ok, nslookup not available in SLE Micro... damn... ok to use wget. But still, the idea of comparing both htmls would be interesting. Not sure how though...

Reply:
@jlausuch what do you think now? For comparing the output we probably need more stable website plus we use HTTP anyways in the other tests (f.e. `zypper` uses it).

Reply:
Yes! And even  `ping` is actually more likely to be installed in any host and container than `wget` :D 
Thanks!